functions/src/modules/Users.functions.ts

```diff
@@ -1,62 +1,162 @@
+  PhoneNumber = 'Phone Number',
+  WhatsApp = 'WhatsApp',
+  Languages = 'Languages',
+  Services = 'Services',
+  Experience = 'Experience',
+  Influencer = 'Influencer',
+  RecommendedBy = 'Recommended By',
+}
+
+enum Roles {
+  CR = 'CR',
+  FC = 'FC',
+  QC = 'QC',
+  TE = 'TE',
+  SE = 'SE',
+  SQR = 'SQR',
+}
+
+enum Decision {
+  Yes = 'Yes',
+  No = 'No',
+}
+
+/**
+ * 1. Scan the Registrations Google Sheet and exports any new users to database
+ * 2. Do not import if user email already exists
+ *
+ */
+export const importUserRegistrationData = functions.https.onRequest(
+  async (req: functions.Request, res: functions.Response) => {
+    const gsheets: GoogleSheet = new GoogleSheet(
+      functions.config().registrations.spreadsheet_id,
+      'Registrations'
+    );
+
+    const registrationRows = await gsheets.getRows();
+
+    const readyForDatabaseUpdate = registrationRows.map((elem: any) => {
+      return {
+        notes: withDefault(elem[RegistrationColumns.Details]),
+        status: elem[RegistrationColumns.Status],
+        timestamp:
+          new Date(elem[RegistrationColumns.Timestamp]).getTime() / 1000,
+        name: elem[RegistrationColumns.Name],
+        location: elem[RegistrationColumns.Country],
+        emailAddress: elem[RegistrationColumns.EmailAddress],
+        phoneNumber: elem[RegistrationColumns.PhoneNumber],
+        isAvailableOnWhatsApp: elem[RegistrationColumns.WhatsApp],
+        languages: elem[RegistrationColumns.Languages]
+          .split(',')
+          .reduce((result: any, language: string) => {
+            result[language.trim()] = true;
+            return result;
+          }, {}),
+        services: elem[RegistrationColumns.Services],
+        roles: {
+          CR: elem[Roles.CR] === Decision.Yes,
+          FC: elem[Roles.FC] === Decision.Yes,
+          QC: elem[Roles.QC] === Decision.Yes,
+          TE: elem[Roles.TE] === Decision.Yes,
+          SE: elem[Roles.SE] === Decision.Yes,
+          SQR: elem[Roles.SQR] === Decision.Yes,
+        },
+        experience: elem[RegistrationColumns.Experience],
+        influencer: elem[RegistrationColumns.Influencer],
+        recommendedBy: elem[RegistrationColumns.RecommendedBy],
+      };
+    });
+
+    await readyForDatabaseUpdate.forEach(async (spreadsheetRecord: any) => {
+      const usersRef = db.ref('/users');
+      await usersRef
+        .orderByChild('emailAddress')
+        .equalTo(spreadsheetRecord['emailAddress'])
+        .once('value', async (snapshot: functions.database.DataSnapshot) => {
+          if (snapshot.exists() === false) {
+            // No such email exist, create new record
+            await usersRef.push(spreadsheetRecord);
+          }
+        });
+    });
+
+    res.status(200).send('Ok');
+  }
+);
+
 /**
  * Process registration of webform data
  * 1. Registers user by transferring data into /users/ path
  * 2. Check if users already exists
- * 
+ *
  * @function processRegistration()
```

Comment:
As soon as you are formatting the whole file, please make this documentation fit the actual function name.

Reply:
Seems now the name has changed to restructureRegistrationData() ?

I'm confused, but I change the comments to reflect this.